kernel/comps/virtio/src/device/block/mod.rs

```diff
@@ -119,15 +119,11 @@ impl VirtioBlockConfig {
         SECTOR_SIZE
     }
 
-    pub(self) fn block_size(&self) -> usize {
-        self.blk_size as usize
+    pub(self) fn block_size(this: &SafePtr<Self, IoMem>) -> usize {
+        field_ptr!(this, Self, blk_size).read_once().unwrap() as usize
     }
 
-    pub(self) fn capacity_sectors(&self) -> usize {
-        self.capacity as usize
-    }
-
-    pub(self) fn capacity_bytes(&self) -> usize {
-        self.capacity_sectors() * Self::sector_size()
+    pub(self) fn capacity_sectors(this: &SafePtr<Self, IoMem>) -> usize {
+        field_ptr!(this, Self, capacity).read_once().unwrap() as usize
```

Comment:
Let's call this method `read_capacity_sectors ` to emphasize the fact that it is not a simple getter.